### PR TITLE
[DX-1414] Tyk Streams Bloblang Query

### PR DESCRIPTION
**Pull request by dcs3spp**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1414](https://tyktech.atlassian.net/browse/DX-1414)

---

### Preview Link

[preview](https://deploy-preview-4855--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-streaming/guides/bloblang/overview)

### Description

Add bloblang supporting documentation for Tyk Streams

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

[DX-1414]: https://tyktech.atlassian.net/browse/DX-1414?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

___

### **PR Type**
Documentation

___

### **Description**
- Added comprehensive documentation for Bloblang functions, including examples and parameters.
- Introduced Bloblang language overview and core features, covering assignments, special characters, non-structured data, deleting fields, variables, metadata, coalescing, literals, comments, boolean logic, arithmetic, conditional mapping, pattern matching, functions, methods, maps, import maps, filtering, and error handling.
- Added advanced Bloblang patterns documentation, including map parameters, walking the tree, message expansion, and creating CSV.
- Added error handling documentation for Tyk Streams, explaining patterns for abandoning on failure, recovering failed messages, logging errors, retrying until success, dropping failed messages, rejecting messages, and routing to a dead-letter queue.
- Added documentation for Bloblang arithmetic, explaining mathematical and comparison operators, number degradation, equality, numerical, and boolean comparisons.
- Added documentation for field paths in Tyk Streams, explaining JSON dot path syntax, special characters, arrays, querying, and setting values.
- Updated menu structure to include new Bloblang documentation sections and added links to error handling and field paths documentation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>functions.md</strong><dd><code>Add Bloblang functions reference documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/functions.md
<li>Added comprehensive documentation for Bloblang functions.<br> <li> Included examples and parameters for each function.<br> <li> Detailed usage of functions like <code>counter</code>, <code>deleted</code>, <code>ksuid</code>, <code>nanoid</code>, <br><code>random_int</code>, <code>range</code>, <code>snowflake_id</code>, <code>throw</code>, <code>ulid</code>, <code>uuid_v4</code>, and more.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-68d2a837abba3c3794c769420ebb8282a08074019a1019728b981d13ef66bdcb">+687/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>about.md</strong><dd><code>Add Bloblang language overview and core features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/about.md
<li>Introduced Bloblang language overview and core features.<br> <li> Explained assignments, special characters, non-structured data, <br>deleting fields, variables, metadata, coalescing, literals, comments, <br>boolean logic, arithmetic, conditional mapping, pattern matching, <br>functions, methods, maps, import maps, filtering, and error handling.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-1f463726815629b95f48cbf5a385c3aefbd325ddd71a17545658b16114e03396">+394/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>advanced.md</strong><dd><code>Add advanced Bloblang patterns documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/advanced.md
<li>Added advanced Bloblang patterns documentation.<br> <li> Included sections on map parameters, walking the tree, message <br>expansion, creating CSV.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-f8ad6bda75b1644646e67f101627d940cd72b2810a473a20533be307f877ae47">+158/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>error-handling.md</strong><dd><code>Add error handling documentation for Tyk Streams</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-streaming/configuration/common-configuration/error-handling.md
<li>Added documentation for error handling in Tyk Streams.<br> <li> Explained patterns for abandoning on failure, recovering failed <br>messages, logging errors, retrying until success, dropping failed <br>messages, rejecting messages, and routing to a dead-letter queue.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-3a1c5680edb3b6909fd8dc13496ea49bfc327f173711df7ed341f6592665b971">+151/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>arithmetic.md</strong><dd><code>Add Bloblang arithmetic documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/arithmetic.md
<li>Added documentation for Bloblang arithmetic.<br> <li> Explained mathematical and comparison operators, number degradation, <br>equality, numerical, and boolean comparisons.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-6e773bc796d8b6e3086028fffc4a5bbd9e5d93b9b1a067aeeb58b9484860807e">+40/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>field-paths.md</strong><dd><code>Add field paths documentation for Tyk Streams</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-streaming/configuration/common-configuration/field-paths.md
<li>Added documentation for field paths in Tyk Streams.<br> <li> Explained JSON dot path syntax, special characters, arrays, querying, <br>and setting values.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-d084a19fccd14bfc5ffbe3fa7e409a5985e96bca3cc6fbb529b4a13a3c9fdf39">+61/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu structure for new Bloblang documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml
<li>Updated menu structure to include new Bloblang documentation sections.<br> <li> Added links to error handling and field paths documentation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4855/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+36/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by dcs3spp** [comment]
Closing draft until ready for review

File: tyk-docs/data/menu.yaml [hunk 1]
```diff
@@ -4000,10 +4000,46 @@ menu:
...
             show: True
+      - title: "Reference Guides"
+        category: Directory
+        show: True
+        menu:
+        - title: "Bloblang"
+          category: Directory
+          show: True
+          menu:
+          - title: "About"
```
Comment (letzya) [inline tyk-docs/data/menu.yaml]: @dcs3spp Any reason we use _about_ and not _overview_? it is more suitable in here?

**Comment by letzya** [comment]
https://deploy-preview-4855--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-streaming/reference/bloblang/overview/

File: tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/about.md [hunk 2]
```diff
@@ -0,0 +1,394 @@
+---
+title: Bloblang
+description: Reference documentation for Bloblang and release notes for all our platforms
+tags: [ "bloblang" ]
+---
+
+Bloblang is a language designed for mapping data of a wide variety of forms. It's a safe, fast and powerful way to perform document mapping within Tyk Streams.
+
+Bloblang is available as a [processor]({{< ref "/product-stack/tyk-streaming/configuration/processors/bloblang" >}}) and it's also possible to use Bloblang queries in [function interpolations]({{< ref "/product-stack/tyk-streaming/configuration/common-configuration/interpolation" >}}).
```
Comment (letzya) [inline tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/about.md]: Shoul we cross ref - so we need a link from https://deploy-preview-4855--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-streaming/configuration/processors/bloblang/ to this

**Review by buger — APPROVED** [review]
Yaara comment makes sense, everything else looks good!

File: tyk-docs/data/menu.yaml (hunk 1 above)
Comment (dcs3spp) [reply tyk-docs/data/menu.yaml]: good call thanks, updated to Overview

File: tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/about.md (hunk 2 above)
Comment (dcs3spp) [reply tyk-docs/content/product-stack/tyk-streaming/reference/bloblang/about.md]: updated with cross refs

**Review by letzya — APPROVED** [review]

**Pull request merged**